src/main/java/org/opensearch/neuralsearch/processor/ScoreNormalizationTechnique.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import lombok.Builder;
+import lombok.Data;
+
+import com.google.common.primitives.Floats;
+
+/**
+ * Collection of techniques for score normalization
+ */
+public enum ScoreNormalizationTechnique {
+
+    /**
+     * Min-max normalization method.
+     * nscore = (score - min_score)/(max_score - min_score)
+     */
+    MIN_MAX {
+        @Override
+        float normalize(ScoreNormalizationRequest request) {
```

Comment:
```suggestion
        public float normalize(final float score, final float minScore, final float maxScore) {
```

Reply:
I actually try to avoid such long list of arguments, as enum abstract method will be implemented by each technique, and most likely every one of it will need different set of params. So instead of adding arguments and pass some mocked values and prefer using DTO, it's more resilient to such changes

Reply:
Sharing single DTO and using different fields for each technique is not a good design pattern because you don't know which values in the DTO is being used for each of technique. If more technique is added, it will get more complex to understand which parameter is for which techniques.

Reply:
I have doubts about that statement, I think in some cases it's a good pattern. I had in mind the idea of sharing same interface across all techniques and for that having single interface is required. In such case approach with each param being an argument has the same cons that you've mentioned - you need to set say 7-8 arguments but not all of them is needed for every technique, and on top of that code is longer which makes it harder to read, and every addition or deletion of an argument requires a change in interface and every implementing class. 
My idea was that the DTO is a base class, and each technique extends it if needed and it knows what params are required. Just in this PR it's a single technique for each of sub-processor so it's not that obvious. Also after Navneet's comments I will probably redesign processor and normalizer as current version is very lean to a single method of min-max.

Reply:
Sharing same interface across multiple techniques could be a better approach. For example
```
public interface ScoreNormalization {
  float normalize(final float score);
}

public class MinMaxNormalization implements ScoreNormalization {
  private final float minScore;
  private final float maxScore;
  
  public float normalize(final float score) {
    //implementation
  }
  
  // Builder of MinMaxNormalization
}
```